### PR TITLE
Add builder methods and move NvPublic

### PR DESCRIPTION
**Pull request by ionut-arm**
This commit brings two changes:
* implementing `builder` methods on all types that have their own
`...Builder` type, so that users are more easily directed to them in the
documentation.
* moving the `nv` top-level module within the `structures` module to
make things a bit more consistent.

Signed-off-by: Ionut Mihalcea <ionut.mihalcea@arm.com>

### CODE REVIEW

File: tss-esapi/src/attributes/nv_index.rs [hunk 1]
```diff
@@ -107,6 +107,11 @@ impl NvIndexAttributes {
 
         Ok(())
     }
+
+    /// Get a builder for the structure
+    pub fn builder() -> NvIndexAttributesBuilder {
```
Comment (Superhepper) [inline tss-esapi/src/attributes/nv_index.rs:112]: Should be made const

**Review by Superhepper — COMMENTED** [review]
You should make the builder functions const were it is possible.

**Comment by ionut-arm** [comment]
> You should make the builder functions const were it is possible.

I've now made all the `builder` methods that I could constant.

**Review by Superhepper — APPROVED** [review]
Nice!

**Review by wiktor-k — APPROVED** [review]
Looks straightforward :+1:

**Pull request merged**